internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -62,7 +77,199 @@ func (r *ClusterDeploymentsReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	return ctrl.Result{}, nil
+	c, err := r.Installer.GetClusterByKubeKey(req.NamespacedName)
+
+	if gorm.IsRecordNotFoundError(err) {
+		return r.createNewCluster(ctx, req.NamespacedName, cluster)
+	}
+
+	return r.updateState(ctx, cluster, c, nil)
+}
+
+func (r *ClusterDeploymentsReconciler) createNewCluster(
+	ctx context.Context,
+	key types.NamespacedName,
+	cluster *hivev1.ClusterDeployment) (ctrl.Result, error) {
+
+	r.Log.Infof("Creating a new cluster %s %s", cluster.Name, cluster.Namespace)
+	spec := cluster.Spec
+
+	pullSecret, err := getPullSecret(ctx, r.Client, spec.PullSecretRef.Name, key.Namespace)
+	if err != nil {
+		r.Log.WithError(err).Error("failed to get pull secret")
+		return ctrl.Result{}, nil
+	}
+
+	clusterParams := &models.ClusterCreateParams{
+		//AdditionalNtpSource:      swag.String(spec.AdditionalNtpSource), // TODO: get from AgentEnvSpec
+		//HTTPProxy:                swag.String(spec.HTTPProxy), // TODO: get from AgentEnvSpec
+		//HTTPSProxy:               swag.String(spec.HTTPSProxy), // TODO: get from AgentEnvSpec
+		//NoProxy:                  swag.String(spec.NoProxy), // TODO: get from AgentEnvSpec
+		//SSHPublicKey:          spec.SSHPublicKey, // TODO: get from AgentEnvSpec
+		BaseDNSDomain:    spec.BaseDomain,
+		Name:             swag.String(spec.ClusterName),
+		OpenshiftVersion: swag.String("4.7"), // TODO: check how to set openshift version
+		Operators:        nil,                // TODO: handle operators
+		PullSecret:       swag.String(pullSecret),
+	}
+
+	if cluster.Spec.InstallStrategy != nil && cluster.Spec.InstallStrategy.Agent != nil {
```

Comment:
If the install strategy isn't Agent then we shouldn't do anything with this cluster - it will be installed by Hive as today and not by this controller.
(This "if" statement should be at the very top of the function)

Reply:
Good point.

Reply:
Added `isSupportedPlatform` that is called just after we get the resource. if the platform is not supported it just return.